src/Concerns/HasMoneyAttributes.php

```diff
@@ -26,8 +27,20 @@ public function getCurrency(): Currency
             return $this->currency;
         }
 
-        if ($this->getRecord()) {
-            return Currency::fromCode($this->getRecord()->{$this->getCurrencyColumn()});
+        $state = match (true) {
+            $this instanceof \Pelmered\FilamentMoneyField\Forms\Components\MoneyInput => $this->getState(),
+            $this instanceof \Pelmered\FilamentMoneyField\Forms\Components\MoneyColumn => $this->getState(),
+            default => null,
+        };
+        if ($state instanceof Money) {
+            return Currency::fromMoney($state);
+        }
```

Comment:
This needs to go before the match. If `$state` is instance of `Money` it will be assigned null in the match statement and this would never trigger. Also generally better to return early.